src/Database/Validator/Permissions.php

```diff
@@ -60,138 +66,138 @@ public function getDescription(): string
             }
 
-            /**
-             * Split role into format {$type}:{$value}
-             *
-             * Substring before ":" $type must be a known permission type
-             * Substring after ":" $value must not be empty and satisty requirements per $type
-             */
-
-            $type = \substr($role, 0, $pos);
-            $value = \substr($role, $pos + 1);
-
-            if (strlen($value) === 0) {
-                $this->message = 'Permission role value must not be empty';
-
+            // Parse permissions string in two parts:
+            //  1. Match method name against known methods, and capture entire permissions strings.
+            //    - Given ["read(users)", "create(user:123abc)", "delete(team:123abc/edit)"]
+            //      - The "read", "create" and delete partitions will be matched against all of $this->methods.
+            //      - Then "users", "user:123abc", "team:123abc/edit" permission strings will be captured.
+            //      - If any invalid method is found, the entire permission parameter will be considered invalid.
+            //  2. Match permission name against known permissions and capture permission name, id and dimension.
+            //    - Given ["users", "user:123abc", "team:123abc/edit"]
+            //      - For each permission string:
+            //        - The "users", "user" and "team" partitions will be matched against all of $this->permissions.
+            //        - Then first iteration will not match any permission ID or dimension.
+            //          - If the permission ID is found, the entire permission string will be considered invalid.
+            //        - Then the second iteration will match "123abc" as a permission ID, with type "user".
+            //          - If the permission ID is not found, the entire permission parameter will be considered invalid.
+            //          - If the permissions ID is not a valid ID, the entire permission parameter will be considered invalid.
+            //        - Then the third iteration will match "123abc" as a permissions ID and "edit" as a dimension.
+            //          - If the permission ID is not found, the entire permission parameter will be considered invalid.
+            //          - If the permissions ID is not a valid ID, the entire permission parameter will be considered invalid.
+            //          - If the dimension should be from the known set and is not found, the entire permission parameter will be considered invalid.
+            //        - If any invalid permission name is found, the entire permission string will be considered invalid.
+
+            // Inner permission string matcher (e.g. "user:123abc", "team:123abc/role") where permission name is matched against known permissions.
+            $permissionMatcher = '((?:' . \implode('|', $this->permissions) . ')(?::(?:[a-z\d]+))?(?:\/(?:[a-z]+))?)';
```

Comment:
we need uppercase and some allowed letters as well right?

```suggestion
            $permissionMatcher = '((?:' . \implode('|', $this->permissions) . ')(?::(?:[a-zA-Z._\-\d]+))?(?:\/(?:[a-zA-Z._\-]+))?)';
```